## Part 2
1. It will print 3. var i was declared in the for loop and it has function scope, so var will be accessible outside of the for loop and its last value where i=3 will be printed.
2. It will print 150 because var discountedPrice has function scope and its last value at the end of [100,200,300] is 300, and when var discountedPrice = prices[2] * (1-.5) was executed, we got var discountedPrice = 300 * .5 = 150.
3. It will print 150 because the last value of discountedPrice, as said in 2., was 150 and when we rounded it with discountedPrice*100/100 == 150*100/100 = 150, we got finalPrice == 150 as finalPrice's last value.
4. It will return an array [50,100,150] because when the function executed it went thru the array [100,200,300], got discountedPrice = price * (1-.5) = .5price for each price, and pushed it into the discounted array.
5. It will throw an error because let i = 0 is block scope (not function scope) so outside of the for loop, i is not defined. As a result, when we try to print i outside of the for loop, it throws an undefined error.
6. It will also throw an error because let discountPrice = {...} has block scope (not function scope) so outside of the for loop, discountPrice is undefined. As a result,  when we try to print discountPRice outside of the for loop, it throws an undefined error.
7. It will print 150. Since finalPrice was declared outside of the for loop in the function scope, finalPrice is accessible anywhere from inside the function. As a result, when finalPrice was changed in the for loop with the last value being passed in from the array of prices as 300 and subsequently discountedPrice = 300 * (.5) == 150, we got finalPrice = 150*100/100 == 150 and when we print finalPrice we get 150.
8. It will return an array [50,100,150]. discounted and finalPrice are declared outside the for loop with function scope, so returning discounted won't throw an undefined error. Furthermore, inside the for loop, we won't get an error because discountedPrice is declared and only used inside the scope of the for loop. Then  we push finalPrice which is half of the values in the array (in the same fashion as described in earlier questions).
9. It will throw an undefined error because i is not defined outside of the scope of the for loop.
10. It will print 3 because length was declared outside of the for loop with function scope, so it is accessible anywhere within the function.
11. It will throw an error because const discounted = [] cannot be changed, which the code inside the for loop discounted.push(discountedPrice) attempts to do.
